extension/jaegerremotesampling/internal/remote_strategy_cache_test.go

```diff
@@ -0,0 +1,247 @@
+	result, ok = cache.get("fooSvc")
+	assert.True(t, ok)
+	assert.Equal(t, testStrategyResponseA, result)
+	result, ok = cache.get("barSvc")
+	assert.False(t, ok)
+	assert.Nil(t, result)
+
+	// reading does not mutate internal cache state
+	assert.Equal(t, serviceStrategyCacheEntry{
+		retrievedAt:      firstWriteTime,
+		strategyResponse: testStrategyResponseA,
+	}, cache.items["fooSvc"])
+
+	// advance time (still within TTL time range)
+	testTime = testTime.Add(20 * time.Millisecond)
+
+	// the written item is still available
+	result, ok = cache.get("fooSvc")
+	assert.True(t, ok)
+	assert.Equal(t, testStrategyResponseA, result)
+	result, ok = cache.get("barSvc")
+	assert.False(t, ok)
+	assert.Nil(t, result)
+
+	// advance time (just before end of TTL time range)
+	testTime = testTime.Add(30 * time.Millisecond)
+
+	// the written item is still available
+
+	result, ok = cache.get("fooSvc")
+	assert.True(t, ok)
+	assert.Equal(t, testStrategyResponseA, result)
+	result, ok = cache.get("barSvc")
+	assert.False(t, ok)
+	assert.Nil(t, result)
+
+	// advance time (across TTL range)
+	testTime = testTime.Add(1 * time.Millisecond)
+
+	// the (now stale) cached item is no longer available
+	result, ok = cache.get("fooSvc")
+	assert.False(t, ok)
+	assert.Nil(t, result)
+	result, ok = cache.get("barSvc")
+	assert.False(t, ok)
+	assert.Nil(t, result)
+
+	assert.Equal(t, serviceStrategyCacheEntry{
+		retrievedAt:      firstWriteTime,
+		strategyResponse: testStrategyResponseA,
+	}, cache.items["fooSvc"])
+}
+
+func Test_serviceStrategyCache_WritesUpdateTimestamp(t *testing.T) {
+	cache := newServiceStrategyCache(cacheTestItemTTL).(*serviceStrategyTTLCache)
+	defer func() {
+		assert.NoError(t, cache.close())
+	}()
+
+	// override clock with a test-controlled one
+	testTime := time.Now()
+	cache.clockFn = func() time.Time {
+		return testTime
+	}
+
+	// initial read returns nothing
+	result, ok := cache.get("fooSvc")
+	assert.False(t, ok)
+	assert.Nil(t, result)
+	result, ok = cache.get("barSvc")
+	assert.False(t, ok)
+	assert.Nil(t, result)
+
+	// perform a write for barSvc at testTime + 10ms
+	testTime = testTime.Add(10 * time.Millisecond)
+	firstWriteTime := testTime
+	cache.put("barSvc", testStrategyResponseA)
+
+	// whitebox assert for internal timestamp tracking
+	assert.Equal(t, serviceStrategyCacheEntry{
+		retrievedAt:      firstWriteTime,
+		strategyResponse: testStrategyResponseA,
+	}, cache.items["barSvc"])
+
+	// read without time advancing
+	result, ok = cache.get("fooSvc")
+	assert.False(t, ok)
+	assert.Nil(t, result)
+	result, ok = cache.get("barSvc")
+	assert.True(t, ok)
+	assert.Equal(t, testStrategyResponseA, result)
+
+	// advance time (still within TTL time range)
+	testTime = testTime.Add(10 * time.Millisecond)
+
+	// the written item is still available
+	result, ok = cache.get("fooSvc")
+	assert.False(t, ok)
+	assert.Nil(t, result)
+	result, ok = cache.get("barSvc")
+	assert.True(t, ok)
+	assert.Equal(t, testStrategyResponseB, result)
```

Comment:
I'm confused: why is this B, and not A?

Reply:
good catch, this is actually the result of my test fixtures being identical (as you pointed out in an earlier comment). fixing this now